Gensim broken with SciPy 1.13.0

Gensim is broken with [SciPy 1.13.0](https://github.com/scipy/scipy/releases/tag/v1.13.0) which was released today.

This is due to the [removal of scipy.linalg.triu](https://github.com/scipy/scipy/pull/18021)

```
.venv/lib/python3.11/site-packages/gensim/__init__.py:11: in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, utils  # noqa:F401
.venv/lib/python3.11/site-packages/gensim/corpora/__init__.py:6: in <module>
    from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
.venv/lib/python3.11/site-packages/gensim/corpora/indexedcorpus.py:14: in <module>
    from gensim import interfaces, utils
.venv/lib/python3.11/site-packages/gensim/interfaces.py:19: in <module>
    from gensim import utils, matutils
.venv/lib/python3.11/site-packages/gensim/matutils.py:20: in <module>
    from scipy.linalg import get_blas_funcs, triu
E   ImportError: cannot import name 'triu' from 'scipy.linalg
```

--- a/gensim/matutils.py
+++ b/gensim/matutils.py
@@ -20,6 +20,10 @@ from scipy.stats import entropy
 from scipy.linalg import get_blas_funcs
 from scipy.linalg.lapack import get_lapack_funcs
 from scipy.special import psi  # gamma function utils
+try:
+    from numpy import triu
+except ImportError:
+    from scipy.linalg import triu
 
 
 logger = logging.getLogger(__name__)
@@ -1127,7 +1131,7 @@ def qr_destroy(la):
     qr, tau, work, info = geqrf(a, lwork=work[0], overwrite_a=True)
     del a  # free up mem
     assert info >= 0
-    r = np.triu(qr[:n, :n])
+    r = triu(qr[:n, :n])
     if m < n:  # rare case, #features < #topics
         qr = qr[:, :m]  # retains fortran order
     gorgqr, = get_lapack_funcs(('orgqr',), (qr,))

